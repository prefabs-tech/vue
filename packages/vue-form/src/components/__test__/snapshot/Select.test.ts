import { mount } from "@vue/test-utils";
import { describe, expect, it } from "vitest";

import MultiSelect from "../../Select.vue";

describe("MultiSelect", () => {
  const options = [
    { value: "FR", label: "FR" },
    { value: "DE", label: "DE" },
    { value: "BE", label: "BE" },
    { value: "FE", label: "FE" },
    { value: "RE", label: "RE" },
  ];

  it("matches snapshot for multiple select", async () => {
    const wrapper = mount(MultiSelect, {
      global: {
        mocks: {
          selectedOptions: [
            { value: "FR", label: "FR" },
            { value: "FE", label: "FE" },
          ],
          showDropdownMenu: true,
        },
      },
      props: {
        label: "Select",
        modelValue: ["FR", "FE"],
        multiple: true,
        options: options,
        placeholder: "Select multiple value",
      },
    });

    expect(wrapper.element).toMatchSnapshot();
  });

  it("matches snapshot for single select", async () => {
    const wrapper = mount(MultiSelect, {
      global: {
        mocks: {
          selectedOptions: [{ value: "FR", label: "FR" }],
        },
      },
      props: {
        label: "Select",
        modelValue: "FR",
        options: options,
        placeholder: "Select a value",
      },
    });

    expect(wrapper.element).toMatchSnapshot();
  });
});
