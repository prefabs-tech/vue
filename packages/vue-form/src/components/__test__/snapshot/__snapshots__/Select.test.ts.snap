// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MultiSelect > matches snapshot for multiple select 1`] = `
<div
  class="multiple-mode multiselect"
  tabindex="0"
>
  <label
    for="multiselect"
  >
    Select
  </label>
  <div
    class="focused multiselect-input"
  >
    <div
      class="field debounce-input multiselect-search"
    >
      <input
        aria-label="input"
        class="input-field"
        placeholder="Select multiple value"
        type="text"
        value=""
      />
    </div>
    <span
      class="action-items"
    >
      <svg
        fill="none"
        tabindex="0"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M6 6L18 18M18 6L6 18"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
        />
      </svg>
      <svg
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          id="Complete"
        >
          <g
            id="F-Chevron"
          >
            <polyline
              fill="none"
              id="down"
              points="5 8.5 12 15.5 19 8.5"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </g>
        </g>
      </svg>
    </span>
  </div>
  <!--teleport start-->
  <!--teleport end-->
</div>
`;

exports[`MultiSelect > matches snapshot for single select 1`] = `
<div
  class="multiselect"
  tabindex="0"
>
  <label
    for="multiselect"
  >
    Select
  </label>
  <div
    class="multiselect-input"
  >
    
    <div
      aria-label="tooltip"
      class="tooltip-container selected-options"
    >
      <div
        class="tooltip-trigger"
      >
        
        
        FR
        
        
      </div>
      <!---->
    </div>
    
    <span
      class="action-items"
    >
      <svg
        fill="none"
        tabindex="0"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M6 6L18 18M18 6L6 18"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
        />
      </svg>
      <svg
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          id="Complete"
        >
          <g
            id="F-Chevron"
          >
            <polyline
              fill="none"
              id="down"
              points="5 8.5 12 15.5 19 8.5"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </g>
        </g>
      </svg>
    </span>
  </div>
  <!--teleport start-->
  <!--teleport end-->
</div>
`;
